discojs/src/client/decentralized/secret_shares.ts

```diff
@@ -19,70 +19,62 @@ export function subtractWeights (w1: Weights, w2: Weights): Weights {
   } return sub
 }
 
-export function sum (setSummands: List<Weights>): Weights { // need to test
-  ''
-  'Return sum of multiple weight objects in an array, returns weight object of sum'
-  ''
-  if (setSummands.size < 1) {
-    return []
-  }
+/*
+Return sum of multiple weight objects in an array, returns weight object of sum
+ */
+export function sum (setSummands: List<Weights>): Weights {
   const summedWeights: Weights = new Array<tf.Tensor>()
   let tensors: Weights = new Array<tf.Tensor>() // list of different sized tensors of 0
-  // @ts-expect-error
-  for (let j = 0; j < setSummands.get(0).length; j++) {
+  const shapeOfWeight: Weights = setSummands.get(0) ?? []
+  for (let j = 0; j < shapeOfWeight.length; j++) {
     for (let i = 0; i < setSummands.size; i++) {
-      // @ts-expect-error
-      tensors.push(setSummands.get(i)[j])
+      const modelUpdate: Weights = setSummands.get(i) ?? []
+      tensors.push(modelUpdate[j])
     }
-    summedWeights.push(tf.addN(tensors))
-    tensors = new Array<tf.Tensor>()
   }
+  summedWeights.push(tf.addN(tensors))
+  tensors = new Array<tf.Tensor>()
   return summedWeights
 }
 
+/*
+Return Weights in the remaining share once N-1 shares have been constructed (where N is number of ready clients)
+ */
 export function lastShare (currentShares: Weights[], secret: Weights): Weights {
-  ''
-  'Return Weights in the remaining share once N-1 shares have been constructed, where N are the amount of participants'
-  ''
   const currentShares2 = List<Weights>(currentShares)
   const last: Weights = subtractWeights(secret, sum(currentShares2))
   return last
 }
 
-export function generateAllShares (secret: Weights, nParticipants: number): List<Weights> {
-  ''
-  'Generate N additive shares that aggregate to the secret array'
-  ''
+/*
+Generate N additive shares that aggregate to the secret weights array (where N is number of ready clients)
+ */
+export function generateAllShares (secret: Weights, nParticipants: number, noiseMagnitude: number): List<Weights> {
   const shares: Weights[] = []
   for (let i = 0; i < nParticipants - 1; i++) {
-    shares.push(generateRandomShare(secret, 1099511627775)) // FIXME random value!
+    shares.push(generateRandomShare(secret, noiseMagnitude))
   }
   shares.push(lastShare(shares, secret))
   const sharesFinal = List<Weights>(shares)
   return sharesFinal
 }
 
-// export function shuffleArray (a: any[]): any[] { // https://stackoverflow.com/questions/6274339/how-can-i-shuffle-an-array
-//   let j, x, i
-//   for (i = a.length - 1; i > 0; i--) {
-//     j = Math.floor(Math.random() * (i + 1))
-//     x = a[i]
-//     a[i] = a[j]
-//     a[j] = x
-//   }
-//   return a
-// }
-
-export function generateRandomNumber (maxRandNumber: number): number {
-  return crypto.randomInt(maxRandNumber)
+/*
+generate a cryptographically secure random number with a maximum value of maxShareValue
+ */
+export function generateRandomNumber (maxShareValue: number): number {
```

Comment:
we probably don't use this one anymore, as it's int not float. also is it signed or not (can be negative?)

Reply:
Does it matter if it's int if we are using it as a seed? It is not signed. When we are using random numbers to populate the generated shares, we are using signed floats within the range (-maxShareValue, maxShareValue).